forest/oak/base.py

```diff
@@ -455,19 +455,22 @@ def preprocess_dates(
         time_min = time_min.replace(tzinfo=from_zone).astimezone(to_zone)
         time_max = datetime.strptime(time_end, fmt)
         time_max = time_max.replace(tzinfo=from_zone).astimezone(to_zone)
-        dates = [date for date in dates if time_min <= date <= time_max]
+        dates2 = [date for date in dates if time_min <= date <= time_max]
+    else:
+        dates2 = dates
```

Comment:
More readable?
```python
if time_start is None and time_end is None:
    dates2 = dates
else:
    ...
```
Also, would it make sense to give `dates2` a more descriptive name?